R/build-reference-index.R

```diff
@@ -31,7 +31,7 @@ data_reference_index_rows <- function(section, index, pkg) {
   rows <- list()
   if (has_name(section, "title")) {
     rows[[1]] <- list(
-      title = markdown_text_inline(section$title),
+      title = markdown_text_inline(section$title, pkg = pkg),
```

Comment:
More of a design question: many of pkgdown's internal functions need the pkg object... but it's still a better idea to pass it rather to store it "somewhere else" like an environment, from where it could be accessed from anywhere? As the latter idea would make things harder to debug?

Reply:
Yeah, in general, I think it's better to be explicit about state rather than storing it in some global object somewhere. Sometimes it's just so much effort to feed it down so many functions to the place that actual need it that I'll use global state, but I usually feel bad about it.